src/main/java/com/example/onlinebookstore/dto/CreateBookRequestDto.java

```diff
@@ -1,13 +1,22 @@
 package com.example.onlinebookstore.dto;
 
+import jakarta.validation.constraints.Min;
+import jakarta.validation.constraints.NotNull;
 import java.math.BigDecimal;
 import lombok.Data;
 
 @Data
 public class CreateBookRequestDto {
+    @NotNull
     private String title;
+    @NotNull
+    @Min(0)
     private BigDecimal price;
+    @NotNull
+    @Min(5)
     private String author;
+    @NotNull
+    @Min(13)
     private String isbn;
     private String description;
     private String coverImage;
```

Comment:
I’m not sure that the min and max annotation is suitable for strings, there is a special annotation for this, also add a maximum length limit as in the table creation file for liquibase. you also need to create an annotation and a validator class for the isbn field

Reply:
Add @Size annos.
Change some with NotBlank.